pysat/tests/test_instruments.py

```diff
@@ -28,23 +28,15 @@
 user_download_dict = {'supermag_magnetometer': ['rstoneback', None]}
 
 
-def safe_data_dir():
-    saved_path = pysat.data_dir
-    if saved_path is '':
-        saved_path = '.'
-    return saved_path
-
-
 def remove_files(inst):
-    # remove any files
-    dir = inst.files.data_path
-    # search for files with instrument file extension
-    target = ''.join(('.', inst.files.files.values[0].split('.')[-1]))
-    for the_file in os.listdir(dir):
-        if (the_file.find(target) != -1):
-            file_path = os.path.join(dir, the_file)
-            if os.path.isfile(file_path):
-                os.unlink(file_path)
+    # remove any files downloaded as part of the unit tests
+    temp_dir = inst.files.data_path
+    for the_file in list(inst.files.files.values):
+        if the_file.rfind('_') > the_file.rfind('.'):
```

Comment:
could you add some comments about what these lines are doing?

Reply:
Adding here and in test_sw.py